packages/compiler/core/parser.ts

```diff
@@ -1356,7 +1355,8 @@ function createParser(code: string | SourceFile, options: ParseOptions = {}): Pa
   ): IdentifierNode {
     if (recoverFromKeyword && isKeyword(token())) {
       error({ code: "reserved-identifier" });
-    } else if (token() !== Token.Identifier) {
+    } else if (token() !== Token.Identifier && token() !== Token.StringLiteral) {
+      // For backward-compatibility, we allow string literals to be used as identifiers, but convert to an identifier node.
```

Comment:
I think this is too permissive, it will allow double quotes in places where they weren't allowed before. We should limit this to the cases where we were accepting stringliteral before. Maybe a wrapper like `parseIdentifierAllowingStringLiteral` or an option paramter to allow string literals?

Reply:
good point, will fix.

Reply:
fixed